src/requestHandler.ts

```diff
@@ -997,7 +997,7 @@ export default class RequestHandler {
 
     for (const file of this.app.vault.getMarkdownFiles()) {
       const cachedContents = await this.app.vault.cachedRead(file);
-      const result = search(cachedContents);
+      const result = search(file.basename + "\n\n" + cachedContents);
```

Comment:
I believe the filename is already included in the response; see line 1017 below.  Am I misunderstanding the result shape, or was there a reason that wasn't sufficient?

Reply:
Thanks for pointing this out! 

1. Should be easy to fix by just adding 3 to the start and end position (first line is headline, then 2 linebreaks)
2. That's a good idea, or we return 0 for start and end and document that this means, the match was in the headline.

I will look into this as soon as I have some spare time :)

Reply:
Sorry for the delay, busy times :)

I made the changes, could you please take a look when you have some time? Thanks!